core/controllers/domain_objects_validator.py

```diff
@@ -86,32 +106,78 @@ def validate_change_dict_for_blog_post(change_dict):
             raise Exception(
                 'Invalid tags provided. Tags not in default tags list.')
 
+    return change_dict
 
-def validate_collection_change(obj):
+
+def validate_collection_change(change_dict):
     # type: (Dict[String, Any]) -> None
     """Validates collection change.
 
     Args:
-        obj: dict. Data that needs to be validated.
+        change_dict: dict. Data that needs to be validated.
+
+    Returns:
+        CollectionChange. Returns CollectionChange object after validation.
     """
     # No explicit call to validate_dict method is necessary, because
     # CollectionChange calls validate method while initialization.
-    collection_domain.CollectionChange(obj)
+    return collection_domain.CollectionChange(change_dict)
 
 
-def validate_email_dashboard_data(data):
+def validate_email_dashboard_data(email_dashboard_data_dict):
     # type: (Dict[String, Optional[Union[bool, int]]]) -> None
-    """Validates email dashboard data.
+    """Validates the email dashboard data dict.
 
     Args:
-        data: dict. Data that needs to be validated.
+        email_dashboard_data_dict: dict. Data that needs to be validated.
+
+    Returns:
+        dict(str, any). Returns email dashboard data dict after validation.
     """
+    # The method returns a dict containing fields of email dashboard
+    # query params. This dict represents the UserQueryParams class, which is a
+    # namedtuple. Hence the fields are validating as a part of schema validation
```

Comment:
Hence the fields of the dict are being validated as ...

Reply:
Thank you for review, will take care of these points while creating another PR.